lib/modules/dosomething/dosomething_api/resources/magick_resource.inc

```diff
@@ -0,0 +1,144 @@
+<?php
+
+function _magick_resource_definition() {
+  $magick_resource = [];
+  $magick_resource['magicks'] = [
+    'operations' => [
+
+      // @Index
+      'index' => [
+        'help' => 'List all the magick.',
+        'file' => [
+          'type' => 'inc',
+          'module' => 'dosomething_api',
+          'name' => 'resources/magick_resource',
+        ],
+        'callback' => '_magick_resource_index',
+        'args' => [
+          [
+            'name' => 'action',
+            'description' => 'The action method to access.',
+            'type' => 'string',
+            'optional' => FALSE,
+            'source' => [
+              'param' => 'action',
+            ],
+          ],
+          [
+            'name' => 'campaign',
+            'description' => 'The id of the specified campaign.',
+            'type' => 'string',
+            'optional' => TRUE,
+            'source' => [
+              'param' => 'campaign',
+            ],
+            'default value' => NULL,
+          ],
+          [
+            'name' => 'campaign_run',
+            'description' => 'The id of the specified campaign run.',
+            'type' => 'string',
+            'optional' => TRUE,
+            'source' => [
+              'param' => 'campaign_run',
+            ],
+            'default value' => NULL,
+          ],
+          [
+            'name' => 'user',
+            'description' => 'The id of the specified user.',
+            'type' => 'string',
+            'optional' => TRUE,
+            'source' => [
+              'param' => 'user',
+            ],
+            'default value' => NULL,
+          ],
+          [
+            'name' => 'key',
+            'description' => 'Custom key to allow automatic campaign signup.',
+            'type' => 'string',
+            'optional' => TRUE,
+            'source' => [
+              'param' => 'key',
+            ],
+            'default value' => NULL,
+          ],
+          [
+            'name' => 'source',
+            'description' => 'Source for the request.',
+            'type' => 'string',
+            'optional' => TRUE,
+            'source' => [
+              'param' => 'source',
+            ],
+            'default value' => NULL,
+          ],
+        ],
+        'access callback' => '_magick_resource_access',
+        'access arguments' => ['index'],
+      ],
+
+    ],
+
+  ];
+
+  return $magick_resource;
+}
+
+/**
+ * Determine whether the method is accessable.
+ *
+ * @return bool
+ */
+function _magick_resource_access($method) {
+  // Temp universal access for now.
+  return TRUE;
+}
+
+/**
+ * Mediate the request and pass along to specfied action if
+ * available.
+ *
+ * @param  string $action
+ * @param  string $campaign
+ * @param  string $user
```

Comment:
We don't do that elsewhere is resources which is why I went w/out the `_id`. In a resource it can never be an object, which I think why in other ***_resource.inc** files we don't use `_id`... I think we later adjust things in the transformers.